bemani/backend/gf/base.py

```diff
@@ -0,0 +1,351 @@
+
+        if oldscore is None:
+            # If it is a new score, create a new dictionary to add to
+            scoredata = ValidatedDict({})
+            raised = True
+            highscore = True
+        else:
+            # Set the score to any new record achieved
+            raised = score > oldscore.points
+            highscore = score >= oldscore.points
+            score = max(oldscore.points, score)
+            scoredata = oldscore.data
+
+        # If we have a combo, replace it
+        scoredata.replace_int('combo', max(scoredata.get_int('combo'), combo))
+        history.replace_int('combo', combo)
+
+        #write our skill points
+        print(skill_perc)
+        print(skill_point)
+        scoredata.replace_int('skill_points', max(scoredata.get_int('skill_points'), skill_point))
+        history.replace_int('skill_points', skill_point)
+
+        #write our skill percent
+        scoredata.replace_int('skill_perc', max(scoredata.get_int('skill_perc'), skill_perc))
+        history.replace_int('skill_perc', skill_perc)
+
+        #did we clear the song?
+        scoredata.replace_int('clear', max(scoredata.get_int('clear'), clear))
+        history.replace_int('clear', clear)
+
+        #did we cheat?
+        #scoredata.replace_int('autoclear', max(scoredata.get_int('autoclear'), autoclear))
+        #history.replace_int('autoclear', autoclear)
+
+        #did we full combo?
+        scoredata.replace_int('fullcombo', max(scoredata.get_int('fullcombo'), fullcombo))
+        history.replace_int('fullcombo', fullcombo)
+
+        #did we excellent full combo?
+        scoredata.replace_int('excellent', max(scoredata.get_int('excellent'), excellent))
+        history.replace_int('excellent', excellent)
+
+        #how did we rank?
+        scoredata.replace_int('result_rank', max(scoredata.get_int('result_rank'), result_rank))
+        history.replace_int('result_rank', result_rank)
+
+        #what was the rate of the combo?
+        scoredata.replace_int('combo_rate', max(scoredata.get_int('combo_rate'), combo_rate))
+        history.replace_int('combo_rate', combo_rate)
+
+        #what was the perfect rate?
+        scoredata.replace_int('perfect_rate', max(scoredata.get_int('perfect_rate'), perfect_rate))
+        history.replace_int('perfect_rate', perfect_rate)
+
+
+        # Look up where this score was earned
+        lid = self.get_machine_id()
+
+        print(songid)
+
+        # GFDM for all versions sends all of the songs
+        # a player played at the end of the round. It doesn't send timestamps
+        # for those songs (Jubeat does). So, if a user plays the same song/chart
+        # more than once in a round, we will end up failing to store the attempt
+        # since we don't allow two of the same attempt at the same time for the
+        # same user and song/chart. So, bump the timestamp by one second and retry
+        # well past the maximum number of songs.
+        now = Time.now()
```

Comment:
Also same comment, this code does literally nothing other than change every single timestamp by 55 seconds.